### ISSUE TITLE
q-table records per page menu jumping around on first click

### ISSUE THREAD

**Issue by ssuess** [quasarframework/quasar#14661]
### What happened?

The q-menu in use in table pagination keeps jumping into position (on first click) when it does not need to.

### What did you expect to happen?

I expect the menu to be stable in this context, there is no need for it to reposition as it does. We are not near the edge of a window for example when this happens.

### Reproduction URL

https://quasar.dev/vue-components/table#basic-usage

### How to reproduce?

1. go to any table example on the page like the first basic one (https://quasar.dev/vue-components/table#basic-usage)
2. click on the menu dropdown next to "Records per page"
3. watch as the menu jumps up and to the left slightly the first time used on the page, but not afterwards.

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar)

### Platforms/Browsers

Firefox, Chrome

### Quasar info output

```Shell
Does not depend on my environment, same thing happens in quasar's own docs.
```

### Relevant log output

```Shell
n/a
```

### Additional context

https://github.com/quasarframework/quasar/discussions/14660

**Comment by ssuess** [comment]
> Hi @ssuess! 👋
> 
> It looks like you provided an invalid or unsupported reproduction URL. Do not use any service other than [Codepen](https://codepen.io), [jsFiddle](https://jsfiddle.net), [StackBlitz](https://stackblitz.com), [Codesandbox](https://codesandbox.io), and [GitHub](https://github.com). Make sure the URL you provided is correct and reachable. You can test it by visiting it in a private tab, another device, etc. Please **edit your original post above** and provide a valid reproduction URL as explained.
> 
> Without a proper reproduction, your issue will have to get closed.
> 
> Thank you for your collaboration. 👏

This is not invalid, the code reproduction is in quasar's own docs.

**Comment by hasanerken** [comment]
I agree with ssuess, just to be more clear,  I copied from the quasar's docs the codepen link.
https://codepen.io/hasanerken/pen/vYrQxLZ?editors=101

**Comment by yusufkandemir** [comment]
@hasanerken it's just an empty link, you should save it to get a proper shareable URL.

@ssuess it's invalid, even if it's in Quasar docs, you should click the "Edit in Codepen" button, save it, and share the URL. It's not invalid as not providing a repro or using something like Pastebin, but it's still not what we are looking for. One of the reasons is that sometimes users send the plain page URL or a section URL, but there are lots of examples in there, so it might cause misunderstandings. Also, it might "accidentally" get fixed or the example gets updated which could hide the actual bug if there is any. By creating a Codepen you basically create a snapshot of that example, with the active Quasar&Vue versions at the time. In this case, the issue is clear and can be reproduced in (almost?) all table examples. So, I'll mark it as a confirmed bug.

**Comment by maggie9595** [comment]
Hi, any updates?

**Comment by muzakon** [comment]
hi, still happens

**Comment by rstoenescu** [comment]
Investigated this in depth. It is a duplicate of #11247 ("Jumping QTable pagination menu") and has been fixed since v2.12.1 (June 2023) by c9e43faf2f12d23935679522ddce5c50271378c2. It was simply never linked back to this issue, so it stayed open.

Root cause, for the record: QTable's records-per-page QSelect uses `options-cover`, so the menu is centered on its anchor using the menu's own measured size. On the very first open, the positioning ran while the option list's virtual scroll had only rendered its assumed-size placeholders, not the real (taller, wider) options. The real items landed a frame later, and the end-of-transition reposition then re-centered the now-larger menu: a small visible snap up and to the left. Subsequent opens reuse the measured sizes, which is why it only ever happened once per page.

The v2.12.1 fix defers the first visible positioning until the menu has real dimensions, so the first paint is already at the final spot.

Verified with automated first-open position sampling against published packages: 2.10.0, 2.11.10 and 2.12.0 jump; 2.12.1 and every later version tested (2.12.2, 2.12.7, 2.13.1, 2.14.7, 2.15.4, 2.16.0, 2.18.5, 2.25.0) do not, including under 20x CPU throttle and on the exact repro from the description (quasar.dev table docs page, currently v2.25.0).

@muzakon your December comment came without a reproduction and I could not reproduce any jump on 2.12.1+. If you still see one on a current version, please open a new issue with a Codepen/StackBlitz pinned to your exact Quasar version.

Closing as fixed in v2.12.1.
